docs/Using Randomness/v2-direct-funding/design.md

```diff
@@ -0,0 +1,127 @@
+>
+> Because the consuming contract directly pays the LINK for the request, the cost is calculated during the request and not during the callback when the randomness is fulfilled. Test your callback function to learn how to correctly estimate the callback gas limit.
+>
+> - If the gas limit is underestimated, the callback fails and the consuming contract is still charged for the work done to generate the requested random values.
+> - If the gas limit is overestimated, the callback function will be executed but your contract is not refunded for the excess gas amount that you paid.
+>
+> Make sure that your consuming contracts are funded with enough LINK tokens to cover the transaction costs. If the consuming contract doesn't have enough LINK tokens, your request will revert.
+
+### Request and Receive Data
+
+#### End-to-end diagram
+
+![VRF v2 direct funding method end to end diagram](/images/vrf/v2-direct-funding-e2e.svg)
+
+Two types of accounts exist in the Ethereum ecosystem:
+
+- EOA (Externally Owned Account): An externally owned account that has a private key and can control a smart contract. Transactions can be initiated only by EOAs.
+- Smart contract: A smart contract that does not have a private key and executes what it has been designed for as a decentralized application.
+
+The Chainlink VRF v2 solution uses both off-chain and on-chain components:
+
+- [VRF v2 Wrapper (on-chain component)](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/VRFV2Wrapper.sol): A wrapper for the VRF Coordinator that provides an interface for consuming contracts.
+- [VRF v2 Coordinator (on-chain component)](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/VRFCoordinatorV2.sol): A contract designed to interact with the VRF node. It emits an event when a request for randomness is made, and then verifies the random number and proof of how it was generated by the VRF Node.
+- VRF Node (off-chain component): Listens for requests by subscribing to the VRF Coordinator event logs and calculates a random number based on the block hash and nonce. The node then sends a transaction to the `VRFCoordinator` including the random number and a proof of how it was generated.
+
+#### Explanation
+
+Requests to Chainlink VRF v2 follow the [Request & Receive Data](/docs/request-and-receive-data/) cycle. The VRF wrapper calls the coordinator to process the request using the following steps:
+
+1. The consuming contract must inherit [VRFV2WrapperConsumerBase](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/VRFV2WrapperConsumerBase.sol) and implement the `fulfillRandomWords` function, which is the _callback VRF function_. Submit your VRF request by calling the `requestRandomness` function in the [VRFV2WrapperConsumerBase](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/VRFV2WrapperConsumerBase.sol) contract. Include the following parameters in your request:
```

Comment:
This should render properly with all the ordered items as `1.` and two tabs for all child elements.